pip/req/req_install.py

```diff
@@ -343,11 +343,18 @@ def setup_py(self):
         assert self.source_dir, "No source dir for %s" % self
         try:
             import setuptools  # noqa
-        except ImportError:
+        except ImportError as e:
+            # Python 2 and 3 variants:
+            if e.args[0] in (
+                    "No module named setuptools",
+                    "No module named 'setuptools'"):
```

Comment:
Apparently since `setuptools` raises an `ImportError` it does not seem to end up in `sys.modules`.